tests/beta/interfaces/test_tape_tf.py

```diff
@@ -785,3 +783,571 @@ def test_sampling(self):
 
         assert res.shape == (2, 10)
         assert isinstance(res, tf.Tensor)
+
+
+@pytest.mark.parametrize(
+    "dev_name,diff_method", [["default.qubit", "finite-diff"], ["default.qubit.tf", "backprop"]]
+)
+class TestQNode:
+    """Same tests as above, but this time via the QNode interface!"""
+
+    def test_execution_no_interface(self, dev_name, diff_method):
+        """Test execution works without an interface"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return expval(qml.PauliZ(0))
+
+        a = tf.Variable(0.1)
+
+        with tf.GradientTape() as tape:
+            res = circuit(a)
+
+        assert circuit.qtape.interface == "autograd"
+
+        # without the interface, the tape simply returns an array of results
+        assert isinstance(res, np.ndarray)
+        assert res.shape == (1,)
+        assert isinstance(res[0], float)
+
+        # without the interface, the tape is unable to deduce
+        # trainable parameters
+        assert circuit.qtape.trainable_params == {0, 1}

```

Comment:
Could there be a way for notifying the user about this? It could be misleading to see an output that doesn't correspond to the true values

Reply:
This isn't user facing however (it is an implementation detail), so I'm a bit hesitant to change the logic 🤔